src/components/MenuDesktop.tsx

```diff
@@ -77,22 +147,93 @@ export class MenuDesktop extends React.Component<MenuDesktopProperties> {
             menuItem.disabled && styles.itemTextDisabled
           ];
 
+          if (!menuItem.submenu) {
+            return (
+              <TouchableOpacity
+                key={index}
+                onPress={() => onMenuPress(menuItem.path || '')}
+                disabled={selected || menuItem.disabled}
+                style={styles.itemContainer}
+              >
+                <Text nativeID={menuItem.nativeID} style={textStyles}>
+                  {menuItem.text}
+                  {menuItem.subText && (
+                    <Text style={{ fontSize: 12, marginLeft: 5 }}>
+                      {menuItem.subText}
+                    </Text>
+                  )}
+                </Text>
+              </TouchableOpacity>
+            );
+          }
+
+          const menuIsOpen = this.state.indexMenuOpened === index;
+
           return (
-            <TouchableOpacity
-              key={menuItem.path}
-              onPress={() => onMenuPress(menuItem.path)}
-              disabled={selected || menuItem.disabled}
-              style={styles.itemContainer}
-            >
-              <Text nativeID={menuItem.nativeID} style={textStyles}>
-                {menuItem.text}
-                {menuItem.subText && (
-                  <Text style={{ fontSize: 12, marginLeft: 5 }}>
-                    {menuItem.subText}
-                  </Text>
-                )}
-              </Text>
-            </TouchableOpacity>
+            <View key={index}>
+              <TouchableOpacity
+                onPress={() => this.handleParentMenuPress(index)}
+                style={styles.submenuTitleContainer}
+              >
+                <Text
+                  nativeID={menuItem.nativeID}
+                  style={[styles.itemText, menuTextStyle, menuItem.textStyle]}
+                >
+                  {menuItem.text}
+                  {menuItem.subText && (
+                    <Text style={{ fontSize: 12, marginLeft: 5 }}>
+                      {menuItem.subText}
+                    </Text>
+                  )}
+                </Text>
+                <FontAwesome
+                  name={menuIsOpen ? 'caret-up' : 'caret-down'}
+                  style={styles.submenuTitleIcon}
+                  size={20}
+                />
+              </TouchableOpacity>
```

Comment:
Existem 3 lugares que aqui no render que renderiza um TouchableOpacity com o Text dentro, o que muda são alguns parâmetros:

1. Quando o item não tem submenu
2. Quando o item é o pai do submenu
3. Para os filhos do submenu

Daria pra criar um componente MenuDesktopItem aqui nesse arquivo mesmo (pode inclusive ser um componente funcional bem simples), que renderiza tudo que há em comum entre os 3 casos e passa via prop o que for diferente (ex: onPress, text, se tem que renderizar ícone, etc...).